### PR TITLE
Implement mr test on GCE: maintainence part

### PR DESCRIPTION
**Pull request by lilyeyes**
Implement mr test on GCE: this is the maintenance part
TEAM-6928 - [AC15+A16]Implement mr test on GCE
  [AC15]: GCE product QA (incl. AC2+AC6+AC7)
  [AC16]: GCE maintenance QA (incl. AC2+AC4+AC5)

Using a new opened ticket to track the failures.
Marked the **failures** to **softfail** as they are know issues in order to **not** block bot auto approval.

Related MR: https://gitlab.suse.de/qa-css/openqa_ha_sap/-/merge_requests/415

- Related ticket: https://jira.suse.com/browse/TEAM-6928
- Needles: NA
- Verification run:
NOTE: I only triggered **sle12** saptune test cases via this branch as there are failures need to be marked as softfail, all **sle15** saptune test cases using master branch as they don't have failure.
15SP4: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=%3A11111%3Atest&groupid=477 (all passed)
15SP3: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP3&build=%3A22222%3Atest-15sp3&groupid=477 (all passed)
15SP2: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP2&build=%3A28146%3Avim&groupid=311 (all passed)
15SP1: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP1&build=%3A28146%3Avim&groupid=300 (passed, we can ignore random Network quota failure)
12SP5: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A55555%3Atest-12sp5&groupid=477 (all passed)
12SP4: https://openqa.suse.de/tests/overview?distri=sle&version=12-SP4&build=%3A66666%3Atest-12sp4&groupid=477 (all passed)

JFYI if without this fix/PR the test results look like this: https://openqa.suse.de/tests/10659520#

### CODE REVIEW

File: tests/sles4sap/saptune/mr_test_run.pm [hunk 1]
```diff
@@ -482,11 +482,24 @@ sub wrapup_log_file {
...
     my $ret = '';
+    my $log = '/tmp/log';
 
-    $ret = script_run "$args";
+    $ret = script_run "$args > $log 2>&1";
     if ($ret) {
-        $result = 'fail';
-        record_soft_failure('Error found:jsc#TEAM-6726');
+        script_run "cat $log";
+        record_soft_failure('Error found:jsc#TEAM-7435');
```
Comment (mpagot) [inline tests/sles4sap/saptune/mr_test_run.pm:490]: Do we always like to have soft failure for any ret != 0 ?

could not be the soft failure only in the else of the "if grep" ?

Comment (lilyeyes) [reply tests/sles4sap/saptune/mr_test_run.pm:490]: > Do we always like to have soft failure for any ret != 0 ?
> 
> could not be the soft failure only in the else of the "if grep" ?

We want to continue the testing when the command's ret!=0 so a softfail is needed here. But the test case will be marked as fail ($result = 'fail';) in the end for some situations.

Comment (lilyeyes) [reply tests/sles4sap/saptune/mr_test_run.pm:490]: FYI: https://openqa.suse.de/tests/10659520# (marked as fail in the end)

Comment (mpagot) [reply tests/sles4sap/saptune/mr_test_run.pm:490]:
```
$ret = script_run "$args > $log 2>&1";
if ($ret) {
    enter_cmd "cat $log";
    if (
            script_run(
"grep -E '(sysstat service.*FAIL.*exited disabled|sysstat \(cronjob\) .*FAIL.*ok|UserTasksMax.*FAIL.*max|UserTasksMax \(sapconf DropIn\).*FAIL.*regular)' $log"
            ))  {
        record_soft_failure('Error found:jsc#TEAM-7435');
        # These are known issues so mark them as 'softfail' not 'fail' for not blocking 'bot' auto approval
        # - sysstat service [FAIL] exited disabled != dead disabled
        # - sysstat (cronjob) [FAIL] ok == no link
        # - UserTasksMax [FAIL] max == 12288
        # - UserTasksMax (sapconf DropIn) [FAIL] regular == missing
    }
    else {
        $result = 'fail';
    }
```

**Review by mpagot — APPROVED** [review]
LGTM

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**